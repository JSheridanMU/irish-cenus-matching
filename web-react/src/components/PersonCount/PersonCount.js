import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import Title from '../Title/Title'
import { useQuery, gql } from '@apollo/client'

const useStyles = makeStyles({
  depositContext: {
    flex: 1,
  },
  navLink: {
    textDecoration: 'none',
  },
})

const GET_COUNT_QUERY = gql`
  query getData($year: String) {
    personCount(year: $year)
    houseCount(year: $year)
  }
`

export default function Deposits(props) {
  const classes = useStyles()
  const { loading, error, data } = useQuery(GET_COUNT_QUERY, {
    variables: {
      year: '/' + props.year + '/',
    },
  })
  if (error) return <p>Error</p>
  return (
    <React.Fragment>
      <Title>{props.year}</Title>
      <Typography component="p" variant="h4">
        {loading ? 'Loading...' : data.personCount}
      </Typography>
      <Typography color="textSecondary" className={classes.depositContext}>
        people found in
      </Typography>
      <Typography component="p" variant="h4">
        {loading ? 'Loading...' : data.houseCount}
      </Typography>
      <Typography color="textSecondary" className={classes.depositContext}>
        households
      </Typography>
    </React.Fragment>
  )
}
