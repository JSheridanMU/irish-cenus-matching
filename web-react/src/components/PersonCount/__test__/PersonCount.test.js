import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client'
import PersonCount from '.././PersonCount'

afterEach(cleanup)

const client = new ApolloClient({
  uri: process.env.REACT_APP_GRAPHQL_URI,
  cache: new InMemoryCache(),
})

test('renders without crashing', () => {
  render(
    <ApolloProvider client={client}>
      <PersonCount year={'1901'} />
    </ApolloProvider>
  )
})

test('renders with temporary content', () => {
  const { getByText } = render(
    <ApolloProvider client={client}>
      <PersonCount year={'1901'} />
    </ApolloProvider>
  )

  getByText('1901')
})

test('matches snapshot', () => {
  const tree = renderer
    .create(
      <ApolloProvider client={client}>
        <PersonCount year={'1901'} />
      </ApolloProvider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
