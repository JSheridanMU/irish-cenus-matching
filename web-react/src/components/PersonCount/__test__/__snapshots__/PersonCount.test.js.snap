// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    1901
  </h2>,
  <p
    className="MuiTypography-root MuiTypography-h4"
  >
    Loading...
  </p>,
  <p
    className="MuiTypography-root makeStyles-depositContext-5 MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    people found in
  </p>,
  <p
    className="MuiTypography-root MuiTypography-h4"
  >
    Loading...
  </p>,
  <p
    className="MuiTypography-root makeStyles-depositContext-5 MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    households
  </p>,
]
`;
